processing/src/main/java/io/druid/segment/IndexMerger.java

```diff
@@ -1474,7 +1474,15 @@ public Rowboat apply(Rowboat lhs, Rowboat rhs)
       Object[] rhsMetrics = rhs.getMetrics();
 
       for (int i = 0; i < metrics.length; ++i) {
-        metrics[i] = metricAggs[i].combine(lhsMetrics[i], rhsMetrics[i]);
+        Object lhsMetric = lhsMetrics[i];
+        Object rhsMetric = rhsMetrics[i];
+        if (lhsMetric == null) {
+          metrics[i] = rhsMetric;
+        } else if (rhsMetric == null) {
+          metrics[i] = lhsMetric;
```

Comment:
Defaulting to 0 is bad, because that's not always the right value for some aggregators, especially min/max and complex aggregators.